rs/pocket_ic_server/src/pocket_ic.rs

```diff
@@ -2428,35 +2422,14 @@ fn new_canister_http_adapter(
     // We don't really use the Socks client in PocketIC as we set `socks_proxy_allowed: false` in the request,
     // but we still have to provide one when constructing the production `CanisterHttp` object
     // and thus we use a reserved (and invalid) proxy IP address.
-    let mut http_connector = HttpConnector::new();
-    http_connector.enforce_http(false);
-    http_connector.set_connect_timeout(Some(Duration::from_secs(2)));
-    let proxy_connector = SocksConnector {
-        proxy_addr: "http://240.0.0.0:8080"
-            .parse()
-            .expect("Failed to parse socks url."),
-        auth: None,
-        connector: http_connector.clone(),
+    let config = HttpsOutcallsConfig {
+        http_connect_timeout_secs: 2,
+        http_request_timeout_secs: 2,
+        socks_proxy: "http://240.0.0.0:8080".to_string(),
```

Comment:
Could this be optional in the config so that we can put `None` here?

Reply:
I see how this can make sense. Let me think a bit more about it. The reason why it is not optional is because all nodes in prod have the socks_proxy defined. This is done because at runtime we don't know if the node is on a system subnet or not